components/plugin_marketplace/marketplace_modal.test.js

```diff
@@ -140,5 +150,17 @@ describe('components/marketplace/', () => {
 
             expect(wrapper).toMatchSnapshot();
         });
+
+        test('Should call for track event when searching', () => {
+            const wrapper = shallow(
+                <MarketplaceModal {...baseProps}/>,
+            );
+
+            wrapper.setState({filter: 'nps'});
+            wrapper.instance().doSearch();
```

Comment:
This isn't a pretty way to test as it's doesn't test the surface of a component.

I tried triggering the search button directly, which worked, but `refs` /seem/ to be `nil` for `wrapper`. That caused https://github.com/mattermost/mattermost-webapp/blob/dd32c9b68e4b54565ba4118e1be67c3d597a0bdf/components/plugin_marketplace/marketplace_modal.js#L151 to fail.

Any advice is appreciated.